subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -50,13 +55,85 @@ static uint8_t mcuboot_buf[CONFIG_APP_MCUBOOT_FLASH_BUF_SZ] __aligned(4);
 static int image_type = DFU_TARGET_IMAGE_TYPE_ANY;
 static char *fota_path;
 static char *fota_host;
+static int current_update_instance = -1;
 static int fota_sec_tag;
+#define MAX_INSTANCE_COUNT CONFIG_LWM2M_FIRMWARE_INSTANCE_COUNT
+static char package_name[MAX_INSTANCE_COUNT][CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_NAME_LENGTH];
+static char package_version[MAX_INSTANCE_COUNT][CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_VERSION_LENGTH];
+
+#define MANUFACTURER_LENGTH 25
+#define HWVERSION_LENGTH 18
+
+struct fota_download_work {
+	struct k_work work;
+	int image_type;
+};
 
 static struct k_work_delayable reboot_work;
-static struct k_work download_work;
+static struct fota_download_work download_work;
 
 void client_acknowledge(void);
 
+static void set_version_and_name_information(void)
+{
+	int ret = 0;
+	char manufacturer[MANUFACTURER_LENGTH] = {0};
+	char hw_version[HWVERSION_LENGTH] = {0};
+
+	ret = modem_info_init();
+	if (ret != 0) {
+		LOG_ERR("Modem info Init error: %d\n\r", ret);
+	}
+	modem_info_string_get(MODEM_INFO_FW_VERSION,
+			      package_version[CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_INSTANCE_MODEM],
+			      MODEM_INFO_MAX_RESPONSE_SIZE);
+
+	ret = nrf_modem_at_scanf("AT+CGMI", "%" STRINGIFY(MANUFACTURER_LENGTH) "[.:0-9A-Za-z ]",
+				 &manufacturer);
+	if (ret != 1) {
+		LOG_ERR("Failed to obtain manufacturer, error: %d", ret);
+	}
+
+	ret = nrf_modem_at_scanf("AT%HWVERSION",
+				 "%%HWVERSION: %" STRINGIFY(HWVERSION_LENGTH) "[.:0-9A-Za-z ]",
+				 &hw_version);
+	if (ret != 1) {
+		LOG_ERR("Failed to obtain hw version, error: %d", ret);
+	}
+	strcat(package_name[CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_INSTANCE_MODEM], "modem:");
```

Comment:
Perhaps add some helper defines like `#define MODEM/APP_INSTNACE CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_INSTANCE_MODEM/APP` in the file to shorten all these expressions, those long symbols makes the code a bit hard to read.